### PR TITLE
Implement cupy.all, cupy.ndarray.all, cupy.any and cupy.ndarray.any

### LINKED ISSUE #409: Implement `any` and `all` in `cupy.ndarray`
To check boolean arrays, we need to make `any` and `all` methods.
related to #403 , #404

### PR DESCRIPTION
**Pull request by delta2323**
This PR implements cupy.all and cupy.any. It also implements corresponding `ndarray`'s methods (`ndarray.all` and `ndarray.any`). 

This PR fixes #409.

### CODE REVIEW

File: cupy/__init__.py [hunk 1]
```diff
@@ -1301,6 +1305,9 @@ def _should_use_rop(self, a):
 equal = logic.comparison.equal
 not_equal = logic.comparison.not_equal
 
+all = logic.truth.all
+any = logic.truth.any
```
Comment (unnonouno) [inline cupy/__init__.py]: This change overwrite built-in `all` and `any` in this source.

Comment (unnonouno) [inline cupy/__init__.py]: It affects this line: https://github.com/pfnet/chainer/pull/493/files#diff-944d22b7359c840a90125a0eaca62545R1447

File: cupy/__init__.py [hunk 2]
```diff
@@ -646,8 +646,12 @@ def prod(self, axis=None, dtype=None, out=None, keepdims=None):
         return prod(self, axis=axis, dtype=dtype, out=out, keepdims=keepdims)
 
     # TODO(okuta): Implement cumprod
-    # TODO(okuta): Implement all
-    # TODO(okuta): Implement any
+
+    def all(self, axis=None, out=None):
+        return all(self, axis, out)
```
Comment (unnonouno) [inline cupy/__init__.py]: Use `logic.truth.all` for readability

File: cupy/__init__.py [hunk 3]
```diff
@@ -646,8 +646,12 @@ def prod(self, axis=None, dtype=None, out=None, keepdims=None):
...
+
+    def any(self, axis=None, out=None):
+        return any(self, axis, out)
```
Comment (unnonouno) [inline cupy/__init__.py]: Use `logic.truth.any` for readability

**Comment by unnonouno** [comment]
I think this test case only tests `True` case. Do we need to test `False` case?

**Comment by delta2323** [comment]
I added the test cases and fixed the critical mistakes.

File: tests/cupy_tests/logic_tests/test_truth.py [hunk 4]
```diff
@@ -1,9 +1,104 @@
...
+
+        c = getattr(xp, self.f)(x, axis, keepdims=True)
+
+        out_shape2 = _calc_out_shape(x.shape, axis, True)
+        d = xp.empty(out_shape2, dtype=x.dtype)
+        getattr(xp, self.f)(x, axis, out=d, keepdims=True)
+        return (a, b, c, d)
+
+    def check(self, x, xp, *axes):
+        return sum([self.check_one_axis(x, xp, a) for a in axes], ())
```
Comment (unnonouno) [inline tests/cupy_tests/logic_tests/test_truth.py]: It is confusing. How about using `for` statement?

**Comment by delta2323** [comment]
I made unit tests of `cupy.all` and `cupy.any` parameterized (See #544). I would like the assignee to check this PR after #550 because this PR uses utility function `_product` which should be in `cupy.testing` module (as I did in #550). If #550 is merged, I will fix this PR to use it.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**